src/main/scala/faces/utils/Helpers.scala

```diff
@@ -209,17 +306,37 @@ case class Helpers(cfg: FacesSettings)(implicit rnd: Random) {
     withLight
   }
 
-  def write(img: PixelImage[RGBA], rps: RenderParameter, id: Int, n: Int): Unit ={
-    if (nSamples > 0){
-      PixelImageIO.write(img.map{f=>f.toRGB}, new File(outImgPath + id + "_" + n + ".jpg"))
-      RenderParameterIO.write(rps, new File(outRpsPath + id + "_" + n + ".rps"))
-      writeCSV(rps, new File(outCSVPath + id + "_" + n + ".csv") )
+  def writeImg(img: PixelImage[RGBA], id: Int, n: Int, postfix: String): Unit = {
+    val outImgPathID = outImgPath + id + "/"
+    if (!Path(outImgPathID).exists) {
+      Path(outImgPathID).createDirectory(failIfExists = false)
+    }
+    PixelImageIO.write(img.map { f => f.toRGB }, new File(outImgPathID + id + "_" + n + postfix + ".png"))
+  }
+
+  def writeExceptImage(rps: RenderParameter, id: Int, n: Int): Unit = {
+    val outRpsPathID= outRpsPath + id + "/"
+    if (!Path(outRpsPathID).exists) {
+      Path(outRpsPathID).createDirectory(failIfExists = false)
     }
-    else {
-      PixelImageIO.write(img.map{f=>f.toRGB}, new File(outImgPath + id + ".jpg"))
-      RenderParameterIO.write(rps, new File(outRpsPath + id +  ".rps"))
-      writeCSV(rps, new File(outCSVPath + id + ".csv"))
+    val outCSVPathID= outCSVPath + id + "/"
+    if (!Path(outCSVPathID).exists) {
+      Path(outCSVPathID).createDirectory(failIfExists = false)
     }
+    val outTLMSPathID= outTLMSPath + id + "/"
+    if (!Path(outTLMSPathID).exists) {
+      Path(outTLMSPathID).createDirectory(failIfExists = false)
+    }
+
+    RenderParameterIO.write(rps, new File(outRpsPathID + id + "_" + n + ".rps"))
+    writeCSV(rps, new File(outCSVPathID + id + "_" + n + ".csv") )
+    writeLandmarks(rps, new File(outTLMSPathID + id + "_" + n + ".tlms"))
+  }
 
+  def write(img: PixelImage[RGBA], rps: RenderParameter, id: Int, n: Int): Unit = {
+    writeExceptImage(rps, id, n)
```

Comment:
Why is here the call to `writeExceptImg`? Looking at call-site the work seems to be made twice.

Reply:
Sorry my mistake.... `write` does both but not `writeImg`.